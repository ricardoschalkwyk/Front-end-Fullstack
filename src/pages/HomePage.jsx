import React, { useEffect, useState } from "react";

import Api from "../Api";

import Search from "../components/Search";

// HomePage holds the Search component and all its children which include display of the search functionality and search results

function HomePage() {
  // This state holds your search input
  const [search, setSearch] = useState({});
  // This state hold your favorited result
  const [favorites, setFavorites] = useState([]);
  // This state is used for loading of search results
  const [isLoading, setIsLoading] = useState(true);

  async function getFavorites() {
    // This function GETS your favorites so that a heart stays red even if you search something else
    try {
      const res = await Api.get("/favorites");
      setFavorites(res);
    } catch (error) {
      console.log(error);
    }
  }
  async function getSearchResults() {
    // This function GETS an initial search value to display how it would look
    try {
      const res = await Api.get("/search?term=Ali%20Gatie&limit=10");
      setSearch(res);
      // Isloading state is changed to false when the results are finished fetching
      setIsLoading(false);
    } catch (error) {
      console.log(error);
    }
  }

  useEffect(() => {
    // This useEffect makes sure that these functions only fire once on render
    getFavorites();
    getSearchResults();
  }, []);

  return (
    <div className="container mx-auto">
      {/* Search Component */}
      <Search
        setIsLoading={setIsLoading}
        setSearch={setSearch}
        search={search}
        favorites={favorites}
        isLoading={isLoading}
      />
    </div>
  );
}

export default HomePage;
