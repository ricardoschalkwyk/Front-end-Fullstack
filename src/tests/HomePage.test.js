import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import "@testing-library/jest-dom";

import HomePage from "../pages/HomePage";

it("renders without crashing", () => {
  const div = document.createElement("div");
  render(<HomePage />, div);
});

it("Matches Snapshot", () => {
  const tree = renderer.create(<HomePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
