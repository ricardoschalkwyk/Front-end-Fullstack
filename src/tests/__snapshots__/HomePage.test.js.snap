// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="container mx-auto"
>
  <div>
    <h1
      className="text-white mb-2 text-lg"
    >
      What would you like to find?
    </h1>
    <div
      className="text-black"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="flex space-x-3"
          >
            <input
              className="rounded border-solid border-1 border-slate-900 py-2 px-3"
              onChange={[Function]}
              placeholder="Search"
              type="text"
              value="Ali Gatie"
            />
            <select
              className="rounded-sm border-solid border-1 border-slate-900 py-2 px-3"
              id=""
              name="select"
              onChange={[Function]}
              value="all"
            >
              <option
                value="movie"
              >
                Movie
              </option>
              <option
                value="podcast"
              >
                Podcast
              </option>
              <option
                value="music"
              >
                Music
              </option>
              <option
                value="musicVideo"
              >
                Music Video
              </option>
              <option
                value="audiobook"
              >
                Audiobook
              </option>
              <option
                value="shortFilm"
              >
                Short Film
              </option>
              <option
                value="tvShow"
              >
                TV Show
              </option>
              <option
                value="software"
              >
                Software
              </option>
              <option
                value="ebook"
              >
                Ebook
              </option>
              <option
                value="all"
              >
                All
              </option>
            </select>
            <select
              className="rounded-sm border-solid border-1 border-slate-900 py-2 px-3"
              name="select"
              onChange={[Function]}
              value="10"
            >
              <option
                value={1}
              >
                1
              </option>
              <option
                value={2}
              >
                2
              </option>
              <option
                value={3}
              >
                3
              </option>
              <option
                value={4}
              >
                4
              </option>
              <option
                value={5}
              >
                5
              </option>
              <option
                value={6}
              >
                6
              </option>
              <option
                value={7}
              >
                7
              </option>
              <option
                value={8}
              >
                8
              </option>
              <option
                value={9}
              >
                9
              </option>
              <option
                value={10}
              >
                10
              </option>
              <option
                value={11}
              >
                11
              </option>
              <option
                value={12}
              >
                12
              </option>
              <option
                value={13}
              >
                13
              </option>
              <option
                value={14}
              >
                14
              </option>
              <option
                value={15}
              >
                15
              </option>
              <option
                value={16}
              >
                16
              </option>
              <option
                value={17}
              >
                17
              </option>
              <option
                value={18}
              >
                18
              </option>
              <option
                value={19}
              >
                19
              </option>
              <option
                value={20}
              >
                20
              </option>
              <option
                value={21}
              >
                21
              </option>
              <option
                value={22}
              >
                22
              </option>
              <option
                value={23}
              >
                23
              </option>
              <option
                value={24}
              >
                24
              </option>
              <option
                value={25}
              >
                25
              </option>
            </select>
            <button
              className="px-3 py-1 bg-slate-500 rounded hover:bg-neutral-900 text-white flex gap-1 items-center"
              onClick={[Function]}
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="w-5 h-5 text-white"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="mt-10 text-white text-base"
    >
      <div
        className="mb-12"
      >
        <div
          className="rounded-md flex flex-col gap-3"
        >
          <div
            className="text-3xl"
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
